packages/core/src/lib/input/range-input.svelte

```diff
@@ -0,0 +1,214 @@
+ * `step`.
+ */
+export let showPips = true;
+
+/** An optional suffix to render after the value, e.g.: % */
+export let suffix: string | undefined = undefined;
+
+/** Additional CSS classes to pass to the input. */
+let extraClasses: cx.Argument = '';
+export { extraClasses as cx };
+
+const indicatorPosition = writable(0);
+const datalistId = uniqueId('range-input');
+let datalist: HTMLDataListElement | undefined;
+
+$: range = max - min;
+$: steps = range / step;
+$: hasExcessivePips = steps >= 100;
+$: excessivePipsStep = steps / 20;
+
+$: suffixValue = suffix ?? '';
+
+$: pipStep = hasExcessivePips ? excessivePipsStep : step;
+$: pipVal = (val: number): number => min + val * step * pipStep;
+
+$: isInputReadOnly = disabled === true || readonly === true;
+$: handleDisabled = preventHandler(isInputReadOnly);
+$: handleDisabledKeydown = preventKeyboardHandler(isInputReadOnly);
+
+$: inputClasses = cx(
+  'slider-track-w-full slider-thumb-w-5 slider-thumb-h-5 slider-thumb-light',
+  'slider-thumb-border slider-track-h-0.5 peer h-7.5 w-full focus:outline-none',
+  !isInputReadOnly &&
+    'slider-track-gray-4 focus:slider-track-gray-5 slider-track-cursor-pointer slider-thumb-cursor-pointer',
+  isInputReadOnly &&
+    'slider-track-disabled-light slider-track-cursor-not-allowed slider-thumb-cursor-not-allowed cursor-not-allowed'
+);
+
+$: indicatorClasses = cx(
+  'invisible absolute bottom-9 -translate-x-1/2 -translate-y-1/2 select-none',
+  'whitespace-nowrap bg-gray-9 px-1.5 py-1 text-center text-xs text-white',
+  'transition duration-200 after:absolute after:-bottom-[2px]',
+  'after:left-[calc(50%-2px)] after:h-0 after:w-0 after:border-x-[2px]',
+  'after:border-b-[0] after:border-t-[2px] after:border-solid',
+  'after:border-x-transparent after:border-b-transparent after:border-t-gray-9',
+  'peer-focus:visible peer-focus:-translate-y-1.5'
+);
+
+let pipCount = 0;
+$: {
+  if (hasExcessivePips && range > excessivePipsStep) {
+    pipCount = range / excessivePipsStep;
+  } else if (hasExcessivePips && range < excessivePipsStep) {
+    pipCount = range / (excessivePipsStep / 100);
+  } else {
+    pipCount = range / step;
+  }
+}
+
+$: pips = Array.from({ length: pipCount + 1 }).map((_, i) => pipVal(i));
+$: positionIndicator = () => {
+  if (!datalist) {
+    return;
+  }
+
+  // Subtracting 20px to account for datalist padding
+  const datalistWidth = datalist.clientWidth - 20;
+
+  // Adding 10px to center on thumb (width of 20px)
+  const left = ((value - min) / (max - min)) * datalistWidth + 10;
+  indicatorPosition.set(left);
+};
+
+/*
+ * TODO (DTCurrie): The below is valid syntax in svelte to invoke a function on a variable change, we should
+ * update our linters to allow this in svelte files, unless we hate it then we can update `positionIndicator`
+ * to accept a `currentValue` argument and pass `value` to it here. That will trigger whenever `value` changes.
```

Comment:
I prefer passing a `currentValue` argument, I think it's easier to understand on first read.